src/data-access/permission/permission.dao.ts

```diff
@@ -0,0 +1,37 @@
+import { PermissionEntity, PermissionEntityModel } from './permission.entity';
+import { NotFoundError } from '../../models';
+import { PermissionDao } from '..';
+
+export class PermissionDaoImpl implements PermissionDao {
+  getById(id: number): Promise<PermissionEntityModel> {
```

Comment:
Is it right return type?

Reply:
yes I decided to return `PermissionEntityModel` instead of `Permission` to have ability perform some db operations on service layer. Yes may be this is not best solution and need to be refactored

Reply:
moved all calls and manipulations with db in dao layer